src/App.js

```diff
@@ -1,25 +1,86 @@
-import logo from './logo.svg';
 import './App.css';
+import React, { useState } from 'react';
+
+export default function App() {
+  const [isValid, setIsValid] = useState(false);
+  const [message, setMessage] = useState('');
+
+  const emailRules = /\S+@\S+\.\S+/;
+  const passwordRules = /^(?=.*\d)(?=.*[!@#$%^&*])(?=.*[a-z])(?=.*[A-Z]).{8,}$/;
+
+  const validateEmail = (event) => {
```

Comment:
Since its a part of learning, could you see to the use of `useCallback` here? This would help to memoize the function and not to recompute it as long as the email does not change. Since the component gets re-rendered every time the props/state change, it would be a easy way to bring some optimization.